daemon/src/model/cfd.rs

```diff
@@ -607,8 +650,66 @@ impl Cfd {
                         transition_timestamp: SystemTime::now(),
                     },
                     dlc,
+                    attestation,
                 }
             }
+            CfdStateChangeEvent::OracleAttestation(attestation) => match self.state.clone() {
+                CfdState::Open { dlc, .. } => CfdState::Open {
+                    common: CfdStateCommon {
+                        transition_timestamp: SystemTime::now(),
+                    },
+                    dlc,
+                    attestation: Some(attestation),
+                },
```

Comment:
We desperately need constructor functions for these states that automatically initialize `common`. This boilerplate makes it really hard to review what is going on.

Reply:
Oh yes, can we tackle that together later?
I looked into how to achieve this in a generic way (given that we have variants with different data inside...) but could not figure it out yet.